lib/src/http/auto_retry.dart

```diff
@@ -170,18 +171,60 @@ class RetryBackOff {
   /// The back-off method to use. One of none, fixed, or exponential.
   final RetryBackOffMethod method;
 
+  /// Whether to enable jitter or not.
+  final bool enableJitter;
+
+  /// The maximum duration between retries.
+  final maxDurationInMs;
+
+  /// The default maximum duration between retries. (5 minutes)
+  static const defaultMaxDurationInMs = 5 * 60 * 1000;
+
+  int _baseBackOffFunction(int numAttempts) {
+    return this.duration.inMilliseconds * pow(2, numAttempts);
+  }
+
+  Duration calculate(int numAttempts) {
+    Random random = new Random();
+    if (enableJitter != null && enableJitter == true) {
```

Comment:
this is unnecessary if `enableJitter` is properly initialized via all constructors. The fixed() and none() constructors are fine since they always initialize `enableJitter` to false, the `exponential()` constructor just needs to account for null. I'll make a comment below